src/components/molecules/KeyValue/KeyValue.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC, ReactElement } from "react";
+import classNames from "classnames";
+
+// Components
+import { IPillProps } from "@components/atoms/Pill";
+import { IKeyProps } from "@components/molecules/KeyValue/Key";
+import { IValueProps } from "@components/molecules/KeyValue/Value";
+
+// Styles
+import "./KeyValue.scss";
+
+interface IKeyValueProps {
+    /**
+     * Additional class for the parent element.
+     * This prop should be used to set placement properties for the element relative to its parent using BEM conventions.
+     */
+    className?: string;
+    /**
+     * Key - value direction <br/>
+     * Possible values: `vertical | horizontal`
+     */
+    direction?: "vertical" | "horizontal";
+    /**
+     * Size
+     * Possible values: `medium | large`;
+     */
+    size?: "medium" | "large";
+    /**
+     * Adds space between key and value in horizontal direction.
+     */
+    spaceBetween?: boolean;
+    children: [ReactElement<IKeyProps>, ReactElement<IValueProps>];
+}
+
+const pillSize: { [key: string]: IPillProps["size"] } = {
+    large: "medium",
+    medium: "small",
+    small: "smallNudge"
+};
+
+export const elementWithType = (
```

Comment:
You can simplify this function.
```suggestion
export const elementWithType = (
    element: JSX.Element,
    type: string,
    keyValueSize: IKeyValueProps["size"] = "medium"
) => {
    const size = element.type.name === "Pill" ? pillSize[keyValueSize] : keyValueSize;
    return element?.type?.name === type ? { ...element, props: { ...element.props, size } } : null;
};
```